### PR TITLE
`docker ps --filter exited=status` should not show running containers

### PR DESCRIPTION
**Pull request by jessfraz**
Docker-DCO-1.1-Signed-off-by: Jessica Frazelle jess@docker.com (github: jfrazelle)

### CODE REVIEW

File: daemon/list.go [hunk 1]
```diff
@@ -73,7 +73,9 @@ func (daemon *Daemon) Containers(job *engine.Job) engine.Status {
 		if !container.Running && !all && n <= 0 && since == "" && before == "" {
 			return nil
 		}
-
+		if len(filt_exited) > 0 && container.Running {
+			return nil
+		}
```
Comment (icecrime) [inline daemon/list.go]: I don't think that's the best fix for the issue. I would do this instead:

```
@@ -98,10 +98,10 @@ func (daemon *Daemon) Containers(job *engine.Job) engine.Status {
                                return errLast
                        }
                }
-               if len(filt_exited) > 0 && !container.Running {
+               if len(filt_exited) > 0 {
                        should_skip := true
                        for _, code := range filt_exited {
-                               if code == container.ExitCode {
+                               if !container.Running && code == container.ExitCode {
                                        should_skip = false
                                        break
                                }
```

Comment (jfrazelle) [inline daemon/list.go]: nice, updated!

**Comment by crosbymichael** [comment]
LGTM

**Comment by vbatts** [comment]
LGTM

**Pull request merged**